pkg/manifests/manifests.go

```diff
@@ -3581,3 +3713,44 @@ func trustedCABundleVolume(configMapName, volumeName string) v1.Volume {
 		},
 	}
 }
+
+func getAlertmanagerTLSSecretsName(alertmanagerConfigs []AdditionalAlertmanagerConfig) []string {
+
+	secretsName := []string{}
+	for _, alertmanagerConfig := range alertmanagerConfigs {
+		if alertmanagerConfig.TLSConfig.CA != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.CA.Name)
+		}
+		if alertmanagerConfig.TLSConfig.Cert != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.Cert.Name)
+		}
+		if alertmanagerConfig.TLSConfig.Key != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.Cert.Name)
+		}
+		if alertmanagerConfig.BearerToken != nil {
+			secretsName = append(secretsName, alertmanagerConfig.BearerToken.Name)
+		}
+	}
+	return removeEmptyDuplicates(secretsName)
+}
+
+func removeEmptyDuplicates(elements []string) []string {
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{}
+	result := []string{}
+
+	for v := range elements {
```

Comment:
you can work on values instead of indices
```suggestion
	for _, v := range elements {
```